### DISCUSSION TITLE
How to traverse pane tree

### DISCUSSION THREAD

**Discussion by AbaoFromCUG** [wezterm/wezterm#5394] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Hyprland

### WezTerm version

20240505-175915-8fa4ba9a

### Ask your question!

In order to save/restore tab layout in lua, I need to traverse pane tree. But I only can get the pane basic information(index, left, top, width, height), without the tree level or others.

I had read the documents:
* https://wezfurlong.org/wezterm/config/lua/MuxTab/index.html
* https://wezfurlong.org/wezterm/config/lua/MuxTab/index.html
* https://wezfurlong.org/wezterm/config/lua/pane/index.html

Thanks

**Comment by wez** [comment]
Take a look at the mux module: https://wezfurlong.org/wezterm/config/lua/wezterm.mux/index.html

**Reply by AbaoFromCUG** [reply]
Oh, I means, I want get the pane binary tree. As you mentioned in https://github.com/wez/wezterm/discussions/3025
> Panes in wezterm are currently implemented as a binary tree, although I may change that in the future.

With `mux` module, I can get all tabs/panes, but I don't known the binary tree of panes. I want to serde/deserde Panes in form of tree, like
```
root(vsplit)
  |- Pane
  |- hsplit
        |-vsplit
        |      |- Pane
        |      |- Pane
        |
        |- Pane
```
Thank you very much for your reply.

**Reply by wez** [reply, marked as answer]
I'm not going to expose this as the binary tree is an implementation detail that I am planning to change
